src/plone/restapi/serializer/dxfields.py

```diff
@@ -73,3 +94,31 @@ def __call__(self):
             'download': url
         }
         return json_compatible(result)
+
+
+@adapter(IObject, IDexterityContent, Interface)
+class ObjectFieldSerializer(DefaultFieldSerializer):
+
+    def check_permission(self, permission_name):
+        return True
+
+    def get_value(self, default=None):
+        result = {}
+        fieldname = self.field.getName()
+        field_value = getattr(self.context, fieldname, None)
+        if field_value:
+            read_permissions = mergedTaggedValueDict(
+                self.field.schema, READ_PERMISSIONS_KEY)
+            for name, field in getFields(self.field.schema).items():
+
+                if not self.check_permission(read_permissions.get(name)):
```

Comment:
Why are you doing the permission checks at all? In your implementation `check_permission()` is always returning true. Thus I would omit all the permission check code (L102-103, L110-111, L114-115).